lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
@kjdelisle here function `create` is actually doing update, when you check the code, here `create` will do a `_insert` with `id` and `_rev` included.